### PR TITLE
Initial landing and project ideas for GSoC 2022

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Initial landing page and the list of project ideas for GSoC 2022. Once landed, it can be improved incrementally. A lot more content to be added but it is enough to kick-off the application.

CC @Stackscribe

Signed-off-by: Oleg Nenashev <o.v.nenashev@gmail.com>

### CODE REVIEW

File: mentorship/gsoc/2022/project-ideas.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+The idea is to extend the current Keptn and Keptn GitOps  operators for more management use-cases
+based on the prototypes created as a part of the Enhancement Proposal.
+Particular details are tio be discussed in the project chat with potential mentors.
+
+- More info: [KEP-67 - Keptn and GitOps](https://github.com/keptn/enhancement-proposals/pull/67)
+- Areas to study/improve: Golang, Kubernetes, Operators, Configuration Management
+- Potential mentor(s): Oleg Nenashev, Thomas Schuetz
+
+### Keptn Plugin for Backstage
```
Comment (oleg-nenashev) [inline mentorship/gsoc/2022/project-ideas.md:25]: Needs to be updated according to the actual state

File: mentorship/gsoc/2022/project-ideas.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+A Possible solution could be a specialized plugin for Keptn using Keptn REST API and/or CLI.
+It might be also possible to create a unified plugin for Keptn and other CI/CD tools, 
+e.g. implemented on the top of the Cloud Events / CDEvents standard. 
+
+
+- More info: https://github.com/keptn/keptn/issues/6407 
+- Areas to study/improve: Golang, JavaScript, React, Backstage
+- Potential mentor(s): Oleg Nenashev, Dmitry Meytin
+
+### Better GitHub Integration in Keptn
```
Comment (oleg-nenashev) [inline mentorship/gsoc/2022/project-ideas.md:43]: More details, or drop it

**Comment by oleg-nenashev** [comment]
Messed up DCO, but can merge still if the text looks good

File: mentorship/gsoc/2022/project-ideas.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
+# GSoC Project Ideas - Keptn
+
+Below you can find Google Summer of Code project ideas
+which have been proposed for this year by Keptn contributors.
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:4]:
```suggestion
This document presents Google Summer of Code project ideas
which Keptn contributors have proposed for this year.
```

File: mentorship/gsoc/2022/project-ideas.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+You can find the application guidelines and more information about Keptn i
+in GSoC on [this page](../README.md)
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:6]:
```suggestion
Application guidelines and more information about Keptn
in GSoC are avaialble on [this page](../README.md)
```

File: mentorship/gsoc/2022/project-ideas.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+<a name="keptn-gitops"></a>
+### GitOps for Keptn
+
+The configuration of a Keptn project requires a lot of imperative commands, executed in the right order.
+
+
+The idea is to extend the current Keptn and Keptn GitOps  operators for more management use-cases
+based on the prototypes created as a part of the Enhancement Proposal.
+Particular details are tio be discussed in the project chat with potential mentors.
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:24]:
```suggestion
Particular details are to be discussed in the project chat with potential mentors.
```

File: mentorship/gsoc/2022/project-ideas.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+- Potential mentor(s): Thomas Schuetz, Oleg Nenashev
+
+<a name="keptn-backstage-plugin"></a>
+### Keptn Plugin for Backstage
+
+[Backstage](https://backstage.io/) is an open platform for building developer portals. 
+As the website states, "powered by a centralized software catalog,
+Backstage restores order to your infrastructure and enables your product teams
+to ship high-quality code quickly — without compromising autonomy".
+It would be nice if it was possible to provide Keptn integration for this portal
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:37]:
```suggestion
It would be nice to provide Keptn integration for this portal
```

File: mentorship/gsoc/2022/project-ideas.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+so that Keptn could be connected as many other tools. See all [Backstage Plugins](https://backstage.io/plugins).
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:38]:
```suggestion
so that Keptn can connected as many other tools. See all [Backstage Plugins](https://backstage.io/plugins).
```

File: mentorship/gsoc/2022/project-ideas.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+- More info: https://github.com/keptn/keptn/issues/6407 
+- Areas to study/improve: Golang, JavaScript, React, Backstage
+
+- Potential mentor(s): Oleg Nenashev, Dmitry Meytin
+
+<a name="keptn-github-integration"></a>
+### Improve GitHub Integration in Keptn
+
+GitHub is widely used in proprietary and open source projects.
+Currently provides several GitHub Actions that could be used to interact with Keptn:
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:54]:
```suggestion
Several GitHub Actions that could be used to interact with Keptn are available:
```

File: mentorship/gsoc/2022/project-ideas.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+especially to provide better user experience for users.
+
+Potential use-cases that could be addressed in the project:
+
+-  **GitHub App Authentication Service** with support for executing GitHub REST API calls with GitHub App authentication.
+   It is required for accessing GitHub's [Checks API](https://docs.github.com/en/rest/reference/checks) and 
+   [Deployments API](https://docs.github.com/en/rest/reference/deployments)
+-  **Support for triggering GitHub Actions** from Keptn so that it can invoke GitHub actions from its sequences and auto-remediation steps.
+   See [Keptn Issue #2670](https://github.com/keptn/keptn/issues/2670) for more details.
+-  **Reporting application statuses to and Deployments API**.
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:67]:
```suggestion
-  **Reporting application statuses to Deployments API**.
```

File: mentorship/gsoc/2022/project-ideas.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+   It would be nice if Keptn could report deployment statuses to GitHub when performing Deployment and auto-remediation sequences.
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:68]:
```suggestion
   It would be nice if Keptn could report deployment status to GitHub when performing Deployment and auto-remediation sequences.
```

File: mentorship/gsoc/2022/project-ideas.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+-  **Reporting Serview Level Objectives(SLOs) and quality gate evaluation results to Checks API**.
+   If Keptn quality gate evaluation is triggered from a GitHub branch/pull request (e.g. staging or preview environment deployment in GitHub Actions based CI/CD),
+   it would be nice if Keptn coud reportevaluation summaries to [Checks API](https://docs.github.com/en/rest/reference/checks)
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:71]:
```suggestion
   it would be nice if Keptn coud report evaluation summaries to the [Checks API](https://docs.github.com/en/rest/reference/checks) 
```

File: mentorship/gsoc/2022/project-ideas.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+Summary:
+
+- Areas to study/improve: Golang, GitHub API, JavaScript
+- Potential mentor(s): Oleg Nenashev
+
+<a name="keptn-k6-integration"></a>
+### Integration service for k6
+
+[k6](https://k6.io/) is a modern cloud natiove tool for performance testing.
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:81]:
```suggestion
[k6](https://k6.io/) is a modern cloud native tool for performance testing.
```

File: mentorship/gsoc/2022/project-ideas.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+It would be nice to support it as a part of Keptn Quality Gates,
+similar to the existing JMeter or Lithmus Chaos services.
+
+- Areas to study/improve: Golang, k6, performance testing
+- Potential mentor(s): TBD
+ 
+ <a name="keptn-jenkins-integration"></a>
+### Jenkins Pipeline Executor Service
+
+Jenkins is widely use as a CI/CD automation tool.
```
Comment (StackScribe) [inline mentorship/gsoc/2022/project-ideas.md:91]:
```suggestion
Jenkins is widely used as a CI/CD automation tool.
```

File: mentorship/gsoc/README.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+## GSoC 2022
+
+- [2022 Project ideas](./2022/project-ideas.md)
+- [Google Summer of Code website](https://summerofcode.withgoogle.com/)
+- [GSoC Timeline](https://summerofcode.withgoogle.com/how-it-works#timeline)
+
+## Participate
+
+We invite GSoC contributors to submit project proposal applications in relation to these ideas.
```
Comment (StackScribe) [inline mentorship/gsoc/README.md:21]: Didn't you say that we would also consider proposals that are unrelated to these ideas?  Maybe add "or other issues that interest you" or something?

File: mentorship/gsoc/README.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+### Apply as a contributor
+
+1. Check out our project ideas.
+2. Join the [Keptn Slack](https://keptn.sh/community/#slack),
+   and the `#gsoc` channel there.
+3. Explore the project ideas and find one or two interesting to you.
```
Comment (StackScribe) [inline mentorship/gsoc/README.md:28]:
```suggestion
3. Explore the project ideas and find one or two that interest you.
```

File: mentorship/gsoc/README.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+   Notify us in the chat.
+4. Join one of our [community or developer meeting](https://keptn.sh/community/meetings/).
```
Comment (StackScribe) [inline mentorship/gsoc/README.md:30]:
```suggestion
4. Join one of our [community or developer meetings](https://keptn.sh/community/meetings/).
```

File: mentorship/gsoc/README.md [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+5. Try contributing to Keptn. Some project ideas include references to newcomer friendly issues, you can also find links to the generic contributing guide [here](https://keptn.sh/community/contributing/)
```
Comment (StackScribe) [inline mentorship/gsoc/README.md:31]:
```suggestion
5. Try contributing to Keptn. Some project ideas include references to newcomer friendly issues. You can also find links to the generic contributing guide [here](https://keptn.sh/community/contributing/)
```

File: mentorship/gsoc/README.md [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+6. Work with the potential mentors during the application phase.
+   Do not hesitate to ask any questions in the chat.
+   It might be possible to also organize special sessions to discuss the project ideas.
+7. Make sure to submit the final application in time!
+
+### Joining as mentor
+
+We welcome contributors to join the mentoring teams.
+Each team, if a project is accepted,
+will consist of 2 or more mentors and one student selected by the mentors
```
Comment (StackScribe) [inline mentorship/gsoc/README.md:41]:
```suggestion
will consist of 2 or more mentors and one participant selected by the mentors 
```
I think that non-students qualify this year.

File: mentorship/gsoc/README.md [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+### Joining as mentor
+
+We welcome contributors to join the mentoring teams.
+Each team, if a project is accepted,
+will consist of 2 or more mentors and one student selected by the mentors 
+and GSoC admins during the application reviews.
+
+**Expectations**. Mentors are expected to be experienced in the project idea domain
+and to be willing to mentor students.
+We also expect mentors to be able to dedicate several hour per week during the community bonding and coding phases.
```
Comment (StackScribe) [inline mentorship/gsoc/README.md:46]:
```suggestion
We expect mentors to be able to dedicate several hour per week during the community bonding and coding phases.
```

File: mentorship/gsoc/README.md [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+
+References:
+
+- [GSoC Mentor Guide](https://google.github.io/gsocguides/mentor/)
+- [Mentor responsibilities](https://developers.google.com/open-source/gsoc/help/responsibilities#mentor_responsibilities)
+- [GSoC Timeline](https://summerofcode.withgoogle.com/how-it-works#timeline)
+
+### Proposing project ideas
+
+To add a new project, submit a pull requests against [this page](./2022/project-ideas.md).
```
Comment (StackScribe) [inline mentorship/gsoc/README.md:56]:
```suggestion
To add a new project idea, submit a pull requests against [this page](./2022/project-ideas.md).
```

**Review by StackScribe — COMMENTED** [review]
lgtm.   I found a couple typos and made some non-essential word-smithing suggestions for your consideration but no blockers here.

File: mentorship/gsoc/README.md [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+2. Join the [Keptn Slack](https://keptn.sh/community/#slack),
+   and the `#gsoc` channel there.
+3. Explore the project ideas and find one or two interesting to you.
+   Notify us in the chat.
+4. Join one of our [community or developer meeting](https://keptn.sh/community/meetings/).
+5. Try contributing to Keptn. Some project ideas include references to newcomer friendly issues, you can also find links to the generic contributing guide [here](https://keptn.sh/community/contributing/)
+6. Work with the potential mentors during the application phase.
+   Do not hesitate to ask any questions in the chat.
+   It might be possible to also organize special sessions to discuss the project ideas.
+7. Make sure to submit the final application in time!
```
[the remark this answers is not in the document]
Comment (oleg-nenashev) [reply mentorship/gsoc/README.md:35]: Will address in a follow-up by adding a new "Key Dates" section

File: mentorship/gsoc/README.md (hunk 14 above)
Comment (oleg-nenashev) [reply mentorship/gsoc/README.md:21]:
```suggestion
We invite GSoC contributors to submit project proposal applications in relation to these ideas.
Original ideas will be welcome and appreciated too!
```

**Comment by oleg-nenashev** [comment]
Will get it merged. so that we work incrementally
